packages/node/src/integrations/anr/worker.ts

```diff
@@ -0,0 +1,214 @@
+  getEnvelopeEndpointWithUrlEncodedAuth,
+  makeSession,
+  updateSession,
+} from '@sentry/core';
+import type { Event, Session, StackFrame, TraceContext, Transport } from '@sentry/types';
+import { callFrameToStackFrame, stripSentryFramesAndReverse, watchdogTimer } from '@sentry/utils';
+import { Session as InspectorSession } from 'inspector';
+import { parentPort, workerData } from 'worker_threads';
+import { makeNodeTransport } from '../../transports';
+import type { WorkerStartData } from './common';
+
+type InspectorSessionNodeV12 = InspectorSession & { connectToMainThread: () => void };
+
+const options: WorkerStartData = workerData;
+let session: Session | undefined;
+let hasSentAnrEvent = false;
+
+function log(msg: string): void {
+  if (options.debug) {
+    // eslint-disable-next-line no-console
+    console.log(`[ANR Worker] ${msg}`);
+  }
+}
+
+const url = getEnvelopeEndpointWithUrlEncodedAuth(options.dsn);
+const transport = makeNodeTransport({
+  url,
+  recordDroppedEvent: () => {
+    //
+  },
+});
+
+async function sendAbnormalSession(): Promise<void> {
+  // of we have an existing session passed from the main thread, send it as abnormal
+  if (session) {
+    log('Sending abnormal session');
+    updateSession(session, { status: 'abnormal', abnormal_mechanism: 'anr_foreground' });
+
+    if (options.debug) {
+      // eslint-disable-next-line no-console
+      console.log(JSON.stringify(session));
+    }
+
+    const envelope = createSessionEnvelope(session, options.dsn, options.sdkMetadata);
+    await transport.send(envelope);
+
+    try {
+      // Notify the main process that the session has ended so the session can be cleared from the scope
+      parentPort?.postMessage('session-ended');
+    } catch (_) {
+      // ignore
+    }
+  }
+}
+
+log('Started');
+
+async function sendAnrEvent(frames?: StackFrame[], traceContext?: TraceContext): Promise<void> {
+  if (hasSentAnrEvent) {
+    return;
+  }
+
+  hasSentAnrEvent = true;
+
+  await sendAbnormalSession();
+
+  log('Sending event');
+
+  const event: Event = {
+    sdk: options.sdkMetadata.sdk,
+    contexts: { ...options.contexts, trace: traceContext },
+    release: options.release,
+    environment: options.environment,
+    dist: options.dist,
+    platform: 'node',
+    level: 'error',
+    exception: {
+      values: [
+        {
+          type: 'ApplicationNotResponding',
+          value: `Application Not Responding for at least ${options.anrThreshold} ms`,
+          stacktrace: { frames },
+          // This ensures the UI doesn't say 'Crashed in' for the stack trace
+          mechanism: { type: 'ANR' },
+        },
+      ],
+    },
+    tags: { 'process.name': 'ANR' },
+  };
+
+  if (options.debug) {
+    // eslint-disable-next-line no-console
+    console.log(JSON.stringify(event));
+  }
+
+  const envelope = createEventEnvelope(event, options.dsn, options.sdkMetadata);
+  await transport.send(envelope);
+  await transport.flush(2000);
```

Comment:
Can you move these magical constant values outside of the function scope, and add a comment for it?